TypeCobol/Compiler/Diagnostics/Diagnostic.cs

```diff
@@ -26,6 +28,8 @@ public Diagnostic(MessageCode messageCode, int columnStart, int columnEnd, int l
                 CatchedException = messageArgs.FirstOrDefault(x => x is Exception) as Exception;
                 MessageArgs = messageArgs;
             }
+
+            _hashCode = (Info==null) ? $"{Line}{ColumnStart}{ColumnEnd}{Message}".GetHashCode() : $"{Line}{ColumnStart}{ColumnEnd}{Message}{Info.Code}{Info.Severity}".GetHashCode();
```

Comment:
There are different technique to calculate the hashCode in this PR:
Sometimes, it combines hash code of different properties and here it use a string concatenation.
What is the best practice ?

Reply:
Use of fields/properties.
The overrides of Equals/GetHashCode, hence IEquatable<Diagnostic>, are not necessary; they have been removed.
https://github.com/TypeCobolTeam/TypeCobol/pull/1555/commits/77c907c8055e95bf877861ca4745d1033758792c